### ISSUE TITLE
Use references instead of loading the whole entity when linking regions, districts and users to other entities

### ISSUE THREAD

**Issue by MartinWahnschaffe** [hzi-braunschweig/SORMAS-Project#10210]
### Problem Description
When saving a core dto, like a CaseDataDto, the system calls the service.getByReferenceDto method for all referenced entities (e.g. UserReferenceDto), which executes a getByUuid method to get the entity. Overall it costs quite some performance (https://github.com/hzi-braunschweig/SORMAS-Project/issues/3481#issuecomment-773260588):

> fillOrBuildEntity setReportingUser: 3ms
> fillOrBuildEntity setClassificationUser: 1ms
> fillOrBuildEntity set jurisdictions: 41ms
> fillOrBuildEntity setSurveillanceOfficer: 9ms
> fillOrBuildEntity setCaseOfficer: 1ms
> fillOrBuildEntity setPointOfEntry: 1ms
> fillOrBuildEntity setReportingDistrict: 0ms

### Proposed Change

Build a UUID-to-ID-mapping cache that allows us to create reference entities that can be used to link an entity to infrastructure entities without having to load them.

- [ ] Create an abstract UuidToIdCache class that can be inherited by a class implementing the singleton pattern, e.g. RegionUuidToIdCache. See UserCache for something similar.
- [ ] Use this cache in RegionService. Do this by overriding the getByUuid method and adding every entity that is retrieved to the cache. When the entity already exists in the map you can use EntityManger.find instead of calling BaseAdoService.getByUuid.
- [ ] Add a getReferenceByUuid method to the RegionService class. It should either use the ID from the cache or query the ID if it's not part of the cache yet. Then use EntityManager.getReference to create a reference entity that is only loaded if needed
- [ ] Override RegionService.getByReferenceDto to use getReferenceByUuid
- [ ] Same for DistrictService
- [ ] Same for UserService

Use the PerformanceLogAnalysisGenerator to create before and after performance logs for saving a case (see https://github.com/hzi-braunschweig/SORMAS-Project/blob/b494beb7e245f65885a7a4187f8c66323a24c0a8/docs/TROUBLESHOOTING.md#debugging-performance-problems). Make sure to save two separate cases that make use of the same users, districts and regions, so we can see the effect of the cache.

Note: The changes above will very likely improve the performance of fillOrBuildEntity for the case, but it might have a negative impact on onCaseChanged and it's sub-method. E.g. setCaseResponsible will have to read the district from the reporting user, which might lead to the reporting user being loaded. This could be mitigated by making the three entity types cacheable (Hibernate L2 cache).

### Alternative Solution A
After #7086 is implemented, use the UUID field as JPA id instead of the Id field. This will mean that all database table references that are currently working by ID would have to be replaced with a UUID. It's not clear what implications this would have on performance.

### Alternative Solution B
Extend ReferenceDto with an id so that when mapping a relation in the DTO-to-Entity mapping we can just create the minimal detached relation entity that gets attached by Hibernate automatically, and don't fetch all the related entities if nothing is being done with them.

This would require us to also adapt the android app and to explain this to external users of the ReST API. It might be quite confusing to them. E.g. SORMAS central provides regions/districts to all SORMAS instances in a country, so they have the same UUID, but they don't have the same ID, so the ID would have to be handled separately for every target system. Also we would have to handle the situation when UUID and ID don't belong to the same entity.
